datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -1010,40 +1010,7 @@ fn aggregate_expressions(
         | AggregateMode::SinglePartitioned => Ok(aggr_expr
             .iter()
             .map(|agg| {
-                let pre_cast_type = if let Some(Sum {
```

Comment:
We no longer need this custom logic, it is handled automatically by the optimizer